src/ExceptionHandling.java

```diff
@@ -32,12 +32,44 @@ public void getLength(String s){
             System.out.println("Exception cause : " + np.getMessage());
         }
     }
+
+    public int getArrayLength(int[] a){
+        try{
+            return a.length;
+        } catch (NullPointerException np){
+            System.out.println("Array is null : " + np.getMessage());
+        }
+        return 0;
+    }
+
+    // Making aware that this API could throw an exception. Possibly to be handled with try catch
+    public void getArrayIndexThrowException(int[] a, int index) throws ArrayIndexOutOfBoundsException {
+        System.out.println(a[index]);
+    }
+
+    //Handling the exception through try catch block
+    public void getArrayIndexHandleException(int[] a, int index) {
```

Comment:
Is it possible to club the exception handling of this method and method on line 36 with a single implementation?

Reply:
Previous function is showing nullpointer exception case whereas this function is showing ArrayIndexOutOfBounds exception. Moreover, it's showing how to handle exception with try catch and in contrast to this, the function at line number 46 is written, where we are throwing the exception in place of handling it with try-catch.

Reply:
Both the exceptions are of the type RuntimeException. Create a separate class which will handle the exception(s) which can be reused in the future.